sklearnex/linear_model/incremental_linear.py

```diff
@@ -0,0 +1,387 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import numbers
+import warnings
+
+import numpy as np
+from sklearn.base import BaseEstimator, MultiOutputMixin, RegressorMixin
+from sklearn.exceptions import NotFittedError
+from sklearn.utils import check_array, gen_batches
+
+from daal4py.sklearn._n_jobs_support import control_n_jobs
+from daal4py.sklearn._utils import sklearn_check_version
+from onedal.linear_model import (
+    IncrementalLinearRegression as onedal_IncrementalLinearRegression,
+)
+
+if sklearn_check_version("1.2"):
+    from sklearn.utils._param_validation import Interval
+
+from onedal.common.hyperparameters import get_hyperparameters
+
+from .._device_offload import dispatch, wrap_output_data
+from .._utils import PatchingConditionsChain, register_hyperparameters
+
+
+@register_hyperparameters(
+    {
+        "fit": get_hyperparameters("linear_regression", "train"),
+        "partial_fit": get_hyperparameters("linear_regression", "train"),
+    }
+)
+@control_n_jobs(
+    decorated_methods=["fit", "partial_fit", "predict", "_onedal_finalize_fit"]
+)
+class IncrementalLinearRegression(MultiOutputMixin, RegressorMixin, BaseEstimator):
```

Comment:
I think you mean removal of sklearn BaseEstimator on onedal4py side. Here it definitely should be, otherwise it would be impossible to run sklearn conformance tests for this estimator.

Reply:
I do not see much use of BaseEstimator in sklearnex/

Reply:
And if so - why is it not present in other incremental algos?

Reply:
yes, you are right. as we decided with @icfaust, sklearn conformance tests should be done for all algos in patch map. BasicStatistics is not in patch map currently (this should be discussed on arch meeting and updated later if necessary). For IncCov it is done in https://github.com/intel/scikit-learn-intelex/pull/1674 and for IncPCA it is unnecessary since it has corresponding class in sklearn

Reply:
thanks for clarification